clients/src/main/java/org/apache/kafka/common/requests/MetadataRequest.java

```diff
@@ -25,24 +25,42 @@
 import java.util.List;
 
 public class MetadataRequest extends AbstractRequest {
-    
+
     private static final Schema CURRENT_SCHEMA = ProtoUtils.currentRequestSchema(ApiKeys.METADATA.id);
     private static final String TOPICS_KEY_NAME = "topics";
 
+    private final boolean allTopics;
     private final List<String> topics;
 
     public MetadataRequest(List<String> topics) {
+        this(topics, false);
+    }
+
+    /**
+     * If allTopics is true, then the passed topics will be ignored.
```

Comment:
Its a bit of a gray area. We introduced the boolean to be explicit and be the single deciding factor for all topics. If we have this validation, its actually the topics list that really matters again and we essentially have 2 toggles to enable all topics.

When validating would we require null & true to get all topics and fail in all other cases in order to match closely to the wire protocols behavior?